File: Java/bfswithhis.java
import java.util.*;

public class BFSwithhis{
    public static void main(String[] args) {
        String[] cities = {"S", "A", "B", "C", "D", "E", "G"};

        int[][] graph = {
            {0,1,1,0,0,0,0},
            {1,0,1,0,1,0,0},
            {1,1,0,1,0,0,0},
            {0,0,1,0,0,1,0},
            {0,1,0,0,0,0,1},
            {0,0,0,1,0,0,0},
            {0,0,0,0,1,0,0}
        };

        String startCit = "S";
        String endCit= "G";

        int startIn = indexOf(cities, startCit);
        int endIn = indexOf(cities, endCit);


        System.out.println("All paths from " + startCit + " to " + endCit + ":");
        bfs(cities, graph, startIn, endIn);
    }
    private static void bfs(String[] cities,int[][] graph,int startIn,int endIn){
        Queue<List<Integer>> queue = new LinkedList<>();
        queue.add(Arrays.asList(startIn));
        
        while(!queue.isEmpty()){
            List<Integer> path = queue.poll();
            int lastnode = path.get(path.size()-1);
            if(lastnode == endIn){
                printpath(path,cities);
            }
            else{
                for(int next = 0;next<cities.length;next++){
                    if(graph[lastnode][next]>0&&!path.contains(next)){
                        List<Integer> newpath = new ArrayList<>(path);
                        newpath.add(next);
                        queue.add(newpath);
                    }
                }
            }
        }
    }
    private static void printpath(List<Integer> path, String[] cities){
        List<String> pathnames = new ArrayList<>();
        
        for(int in:path){
            pathnames.add(cities[in]);
        }
        System.out.println(String.join("->",pathnames));
    }
    
    private static int indexOf(String[] arr,String city){
        for(int i=0;i<arr.length;i++){
            if(arr[i].equals(city)) return i;
        }
        return -1;
    }

}

File: C++/bfswithhis.cpp
#include <iostream>
#include <vector>
#include <string>
#include <queue>
#include <list>
#include <algorithm>

int indexOf(const std::vector<std::string>& arr, const std::string& city) {
    for (size_t i = 0; i < arr.size(); ++i) {
        if (arr[i] == city) {
            return i;
        }
    }
    return -1;
}

void printPath(const std::vector<int>& path, const std::vector<std::string>& cities) {
    for (size_t i = 0; i < path.size(); ++i) {
        std::cout << cities[path[i]];
        if (i < path.size() - 1) {
            std::cout << "->";
        }
    }
    std::cout << std::endl;
}

void bfs(const std::vector<std::string>& cities, const std::vector<std::vector<int>>& graph, int startIn, int endIn) {
    std::queue<std::vector<int>> queue;
    std::vector<int> startPath;
    startPath.push_back(startIn);
    queue.push(startPath);

    while (!queue.empty()) {
        std::vector<int> path = queue.front();
        queue.pop();

        int lastNode = path.back();

        if (lastNode == endIn) {
            printPath(path, cities);
        } else {
            for (size_t next = 0; next < cities.size(); ++next) {
                if (graph[lastNode][next] > 0 &&
                    std::find(path.begin(), path.end(), next) == path.end()) {
                    
                    std::vector<int> newPath = path;
                    newPath.push_back(next);
                    queue.push(newPath);
                }
            }
        }
    }
}

int main() {
    std::vector<std::string> cities = {"S", "A", "B", "C", "D", "E", "G"};

    std::vector<std::vector<int>> graph = {
        {0, 1, 1, 0, 0, 0, 0},
        {1, 0, 1, 0, 1, 0, 0},
        {1, 1, 0, 1, 0, 0, 0},
        {0, 0, 1, 0, 0, 1, 0},
        {0, 1, 0, 0, 0, 0, 1},
        {0, 0, 0, 1, 0, 0, 0},
        {0, 0, 0, 0, 1, 0, 0}
    };

    std::string startCit = "S";
    std::string endCit = "G";

    int startIn = indexOf(cities, startCit);
    int endIn = indexOf(cities, endCit);

    if (startIn != -1 && endIn != -1) {
        std::cout << "All paths from " << startCit << " to " << endCit << ":" << std::endl;
        bfs(cities, graph, startIn, endIn);
    } else {
        std::cout << "Start or end city not found." << std::endl;
    }

    return 0;
}